### PR TITLE
feat: add Parlia genesis config for BSC

### PR DESCRIPTION
**Pull request by forcodedancing**
## Motivation

BSC uses Parlia consensus engine. To support BSC genesis, Parlia config is added.

## Solution
Update genesis config to support Parlia.

### CODE REVIEW

File: crates/genesis/src/lib.rs [hunk 1]
```diff
@@ -423,6 +423,10 @@ pub struct ChainConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub clique: Option<CliqueConfig>,
 
+    /// Parlia parameters.
```
Comment (mattsse) [inline crates/genesis/src/lib.rs:426]: can we add an url to to a spec if that exists?

Comment (forcodedancing) [reply crates/genesis/src/lib.rs:426]: thanks a lot @mattsse 
The introduction and code specification are added.

**Comment by prestwich** [comment]
given that the config props are identical to clique config, should we reuse the struct?

**Comment by forcodedancing** [comment]
> given that the config props are identical to clique config, should we reuse the struct?

You are right. The fields are the same. However, the meanings of the fields are different for clique and parlia. 

I am ok if you think reuse is better. thanks a lot.

**Comment by prestwich** [comment]
> You are right. The fields are the same. However, the meanings of the fields are different for clique and parlia.

Ya, I understand that in Parlia the fields are used differently than in clique. But given that the names and comments are the same it seems safe to reuse. I don't have a strong opinion here, so I'm fine leaving it as 2 identical structs

File: crates/genesis/src/lib.rs [hunk 2]
```diff
@@ -541,6 +545,21 @@ pub struct CliqueConfig {
     pub epoch: Option<u64>,
 }
 
+/// Consensus configuration for Parlia.
+/// Parlia is the consensus engine for BNB Smart Chain.
+/// For the general introduction: https://docs.bnbchain.org/docs/learn/consensus/
```
Comment (prestwich) [inline crates/genesis/src/lib.rs:550]: doc nit again: 
for links you should follow recommendation here:

https://doc.rust-lang.org/rustdoc/lints.html?highlight=http%3A%2F%2F#bare_urls

Comment (DaniPopes) [reply crates/genesis/src/lib.rs:550]: https://github.com/alloy-rs/alloy/actions/runs/9062315958/job/24951891283?pr=740

Comment (forcodedancing) [reply crates/genesis/src/lib.rs]: fixed, thanks a lot

**Review by mattsse — APPROVED** [review]
this doesn't have any impact on how genesis is used for eth and alike so I'm fine with adding this field out of convenience

**Review by prestwich — APPROVED** [review]

**Pull request merged**